test_outputs/inherent_associated_pub_const_missing.output.ron

```diff
@@ -33,5 +33,16 @@
             "span_filename": String("src/lib.rs"),
             "visibility_limit": String("public"),
         },
+        {
+            "associated_constant": String("N"),
+            "name": String("Example"),
+            "path": List([
+                String("inherent_associated_pub_const_missing"),
+                String("Example"),
+            ]),
+            "span_begin_line": Uint64(87),
+            "span_filename": String("src/lib.rs"),
+            "visibility_limit": String("public"),
+        },
```

Comment:
Oooh interesting, this is a false-positive — we wanted this to _not_ get flagged.

Perhaps because the `N` constant isn't defined in the `impl` block and is instead a default item on the trait? 🤔

Sorry this is turning into a bit of a rabbit hole. I see two options to move forward here.

If this PR isn't something you find that interesting and you don't want to go down the rabbit hole (which is totally fine!), we can merge it as-is and I can open a separate issue for resolving this false-positive.

If you find this PR interesting and you'd like to continue polishing it, then it would be great to do the following:
- Add a new test case, similar to the trait+struct one we have already, but where the `const` doesn't have a default value and is instead given a value in the `impl` block for the trait.
- Add comments on both the old and new test cases, to explain what they test, how they are different, and why it's important to have both of them and not just one or the other.
- Depending on how this new test case interacts with the query (whether it gets flagged or not), we'd continue tweaking the query -- for example, by checking two separate cases explicitly: "inherent impl must not have the constant" and "impl of a trait, where that trait must not have a const by that name either."

Your call! Let me know how you'd prefer to move forward.

Reply:
I think I would keep working on it but if you think this is a rare case and it would be more beneficial to have the lint as it is currently than to not have it at all you can merge it. I don't have a problem either way.

Reply:
Nice! There's no rush to merge this if you are happy to keep working on it. I was just offering it in case you felt frustrated by the back-and-forth on it, or if you just wanted a change of pace to something else.

It's great that the new test case didn't trigger the lint. Let's try tweaking the lint then.

We probably want to go back to `inherent_impl` like we had before. Then we probably want to add another block that also says "no implemented trait has a const by that name either." Here's a short snippet in the playground to help you get started: [link](https://play.predr.ag/rustdoc#?f=2&q=query---0Crate---2item---4impl---6implemented_trait---8*3-continue-from-here*l--_6--*4--*2--*0*J*l*J&v=*C*l*l*J)

Lmk if you get stuck or would like another pair of eyes on something!